src/app/api/tasks/tasks.service.ts

```diff
@@ -529,24 +541,29 @@ export class TasksService extends BaseService {
 
     // Case 3
     // --- If task was previously in another state, and is moved to a 'completed' type WorkflowState by IU
+    let shouldCreateNotification = true
     if (
       prevTask?.workflowState?.type !== StateType.completed &&
       updatedTask?.workflowState?.type === StateType.completed &&
       updatedTask.assigneeId
     ) {
       // Don't send task notifications if the IU created the task themselves
       if (updatedTask.createdById === this.user.internalUserId) {
-        return
+        shouldCreateNotification = false
       }
 
       if (updatedTask.assigneeType === AssigneeType.internalUser) {
-        await notificationService.create(NotificationTaskActions.CompletedByIU, updatedTask, { email: true })
+        shouldCreateNotification &&
+          (await notificationService.create(NotificationTaskActions.CompletedByIU, updatedTask, { email: true }))
         // TODO: Clean code and handle notification center notification deletions here instead
       } else if (updatedTask.assigneeType === AssigneeType.company) {
         // Don't do this in parallel since this can cause rate-limits, each of them has their own bottlenecks for avoiding ratelimits
-        await notificationService.create(NotificationTaskActions.CompletedForCompanyByIU, updatedTask, { email: true })
+        shouldCreateNotification &&
+          (await notificationService.create(NotificationTaskActions.CompletedForCompanyByIU, updatedTask, { email: true }))
```

Comment:
so just to understand with the context:
> Don't send task notifications if the IU created the task themselves

And if the IU is the assignee, and s/he completes the task, then do we need to send the notification to himself/herself? or is it skipped in the notificationService?

Reply:
When a notification is completed by IU, the notification are sent to task creator.

In the case that task creator is the IU themselves, this is skipped by the `shouldCreateNotification` initialization
![image](https://github.com/user-attachments/assets/13c578e8-8456-4ef1-a7a4-c8e5b791f6fe)

Else the notification is sent, but the creator is confirmed not to be the IU themselves